### PR TITLE
Exceptions driver

### PR DESCRIPTION
**Pull request by jim-parry**
Exception handling for PDO driver.

### CODE REVIEW

File: system/database/drivers/pdo/pdo_driver.php [hunk 1]
```diff
@@ -109,7 +110,7 @@ public function __construct($params)
 
 			if ($this->db_debug)
 			{
-				show_error('Invalid or non-existent PDO subdriver');
+				throw new RuntimeException('PDO_driver: Invalid or non-existent PDO subdriver');
```
Comment (narfbg) [inline system/database/drivers/pdo/pdo_driver.php]: 'Database' instead of 'PDO_driver'

File: system/libraries/Driver.php [hunk 2]
```diff
@@ -108,7 +109,7 @@ public function load_driver($child)
 			// The requested driver isn't valid!
 			$msg = 'Invalid driver requested: '.$child_name;
```
Comment (narfbg) [inline system/libraries/Driver.php]: This needs to be prefixed with 'Database' too, and it should be hard-coded into the `RuntimeException` throw ...

Exceptions are logged, and therefore `log_message()` should be removed.

File: system/libraries/Driver.php [hunk 3]
```diff
@@ -178,7 +179,7 @@ public function load_driver($child)
 			{
 				$msg = 'Unable to load the requested driver: '.$class_name;
 				log_message('error', $msg);
-				show_error($msg);
+				throw new RuntimeException($msg);
```
Comment (narfbg) [inline system/libraries/Driver.php]: Same as on line 110

**Comment by jim-parry** [comment]
Restarting this, since the feature/exceptions branch is too stale.

**Pull request closed without merging**